mavlink/tests/serde_test.rs

```diff
@@ -236,3 +236,39 @@ mod serde_test {
         );
     }
 }
+
+mod serde_test_json {
+    use mavlink::common;
+
+    #[test]
+    fn test_serde_output() {
+        let json = serde_json::to_string(&common::MavMessage::HEARTBEAT(common::HEARTBEAT_DATA {
+            custom_mode: 0,
+            mavtype: common::MavType::MAV_TYPE_GENERIC,
+            autopilot: common::MavAutopilot::MAV_AUTOPILOT_GENERIC,
+            base_mode: common::MavModeFlag::MAV_MODE_FLAG_SAFETY_ARMED,
+            system_status: common::MavState::MAV_STATE_UNINIT,
+            mavlink_version: 3,
+        }))
+        .unwrap();
+        assert_eq!(json,  "{\"type\":\"HEARTBEAT\",\"custom_mode\":0,\"mavtype\":{\"type\":\"MAV_TYPE_GENERIC\"},\"autopilot\":{\"type\":\"MAV_AUTOPILOT_GENERIC\"},\"base_mode\":\"MAV_MODE_FLAG_SAFETY_ARMED\",\"system_status\":{\"type\":\"MAV_STATE_UNINIT\"},\"mavlink_version\":3}");
```

Comment:
```suggestion
        let expected = json!({
            "type": "HEARTBEAT",
            "custom_mode": 0,
            "mavtype": { "type": "MAV_TYPE_GENERIC" },
            "autopilot": { "type": "MAV_AUTOPILOT_GENERIC" },
            "base_mode": "MAV_MODE_FLAG_SAFETY_ARMED",
            "system_status": { "type": "MAV_STATE_UNINIT" },
            "mavlink_version": 3
        });
        assert_eq(json, expected);
```

Reply:
nice!

Reply:
This is not possible since the order of the parameters change between both approaches